Remove lowered execution

zdevito opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #21814 script::Module as a view.
* #21675 Make script::Method a value type
* **#21674 Remove lowered execution**

Summary: First class modules can now run on our entire test suite,
and benchmarks indicate there is no obvious performance regressions.
This PR removes lowered execution and cleans up dead code without
changing the Module API. A followup PR further simplifies the Module API
now that methods are simply a pair of (function, method_object).

Differential Revision: [D15777726](https://our.internmc.facebook.com/intern/diff/D15777726)

zdevito committed:
Remove lowered execution

Summary: First class modules can now run on our entire test suite,
and benchmarks indicate there is no obvious performance regressions.
This PR removes lowered execution and cleans up dead code without
changing the Module API. A followup PR further simplifies the Module API
now that methods are simply a pair of (function, method_object).

zdevito committed:
Update on "Remove lowered execution"

Remove lowered execution

Summary: First class modules can now run on our entire test suite,
and benchmarks indicate there is no obvious performance regressions.
This PR removes lowered execution and cleans up dead code without
changing the Module API. A followup PR further simplifies the Module API
now that methods are simply a pair of (function, method_object).

gh-metadata: pytorch pytorch 21674 gh/zdevito/63/head

zdevito committed:
Update on "Remove lowered execution"

Remove lowered execution

Summary: First class modules can now run on our entire test suite,
and benchmarks indicate there is no obvious performance regressions.
This PR removes lowered execution and cleans up dead code without
changing the Module API. A followup PR further simplifies the Module API
now that methods are simply a pair of (function, method_object).

gh-metadata: pytorch pytorch 21674 gh/zdevito/63/head

zdevito committed:
Update on "Remove lowered execution"

Remove lowered execution

Summary: First class modules can now run on our entire test suite,
and benchmarks indicate there is no obvious performance regressions.
This PR removes lowered execution and cleans up dead code without
changing the Module API. A followup PR further simplifies the Module API
now that methods are simply a pair of (function, method_object).

gh-metadata: pytorch pytorch 21674 gh/zdevito/63/head

zdevito committed:
Update on "Remove lowered execution"

Remove lowered execution

Summary: First class modules can now run on our entire test suite,
and benchmarks indicate there is no obvious performance regressions.
This PR removes lowered execution and cleans up dead code without
changing the Module API. A followup PR further simplifies the Module API
now that methods are simply a pair of (function, method_object).

gh-metadata: pytorch pytorch 21674 gh/zdevito/63/head

suo commented on torch/csrc/jit/script/module.h:
```diff
@@ -113,32 +103,31 @@ struct TORCH_API Method {
   }
-  const std::string& name() const {
-    return name_;
+  const std::string& field_name() const {
```
field_name is a little confusing—unqualified_name?

zdevito replied:
There are now two names, the name of the Class and the name of the module as a member of a parent module. We still rely on this "field_name" in a few places and it is never qualified.  There is a comment that explains this. My hope is that the "Module because a view patch" simply removes it.

suo commented on aten/src/ATen/core/jit_type.h:
```diff
@@ -1372,11 +1372,7 @@ struct CAFFE2_API ClassType : public Type {
-  // Create a type representing a Module,
-  // These do not have methods, and are not globally registered
-  static ClassTypePtr createModuleType(std::shared_ptr<CompilationUnit> module);
+      std::shared_ptr<CompilationUnit> cu, bool is_module = false);
```
this `is_module` flag is a suspicious to me, since it implies that we are branching somewhere on whether something is a module (and so modules are not quite first class actually)

suo replied:
Ah, seems like it is used for ONNX…

zdevito replied:
Discussed in person: ModuleType will be introduced as subtype of ClassType in a future comment. This means that the ClassType has additional module-like behavior (e.g. it has a list of "parameter" objects)

zdevito committed:
Update on "Remove lowered execution"

Remove lowered execution

Summary: First class modules can now run on our entire test suite,
and benchmarks indicate there is no obvious performance regressions.
This PR removes lowered execution and cleans up dead code without
changing the Module API. A followup PR further simplifies the Module API
now that methods are simply a pair of (function, method_object).

gh-metadata: pytorch pytorch 21674 gh/zdevito/63/head

zdevito committed:
Update on "Remove lowered execution"

Remove lowered execution

Summary: First class modules can now run on our entire test suite,
and benchmarks indicate there is no obvious performance regressions.
This PR removes lowered execution and cleans up dead code without
changing the Module API. A followup PR further simplifies the Module API
now that methods are simply a pair of (function, method_object).

gh-metadata: pytorch pytorch 21674 gh/zdevito/63/head

zdevito committed:
Update on "Remove lowered execution"

Remove lowered execution

Summary: First class modules can now run on our entire test suite,
and benchmarks indicate there is no obvious performance regressions.
This PR removes lowered execution and cleans up dead code without
changing the Module API. A followup PR further simplifies the Module API
now that methods are simply a pair of (function, method_object).

gh-metadata: pytorch pytorch 21674 gh/zdevito/63/head

zdevito committed:
Update on "Remove lowered execution"

Remove lowered execution

Summary: First class modules can now run on our entire test suite,
and benchmarks indicate there is no obvious performance regressions.
This PR removes lowered execution and cleans up dead code without
changing the Module API. A followup PR further simplifies the Module API
now that methods are simply a pair of (function, method_object).

gh-metadata: pytorch pytorch 21674 gh/zdevito/63/head

zdevito committed:
Update on "Remove lowered execution"

Remove lowered execution

Summary: First class modules can now run on our entire test suite,
and benchmarks indicate there is no obvious performance regressions.
This PR removes lowered execution and cleans up dead code without
changing the Module API. A followup PR further simplifies the Module API
now that methods are simply a pair of (function, method_object).

gh-metadata: pytorch pytorch 21674 gh/zdevito/63/head

zdevito committed:
Update on "Remove lowered execution"

Remove lowered execution

Summary: First class modules can now run on our entire test suite,
and benchmarks indicate there is no obvious performance regressions.
This PR removes lowered execution and cleans up dead code without
changing the Module API. A followup PR further simplifies the Module API
now that methods are simply a pair of (function, method_object).

gh-metadata: pytorch pytorch 21674 gh/zdevito/63/head
